package.json

```diff
@@ -58,5 +58,6 @@
   },
   "dependencies": {
     "chalk": "^2.3.0"
-  }
+  },
+  "types": "index.d.ts"
```

Comment:
No need to put this. When the file is named `index.d.ts` and it lives in the root of the project, then TypeScript automatically resolves it even if there is no `types` entry.

Reply:
True, but it is advisable to do so :wink:

Reply:
Can you give some input why it is advisable? Following the software design paradigm of [Convention over Configuration](https://en.wikipedia.org/wiki/Convention_over_configuration) it is not. 

The "types" property is a custom property which is not specified in the official [package.json](https://docs.npmjs.com/files/package.json) documentation, so we should avoid custom properties wherever we can to be standard-compliant.

Reply:
> The "types" property is a custom property which is not specified in the official package.json documentation [...]

So is the `jest` property in [line 49](https://github.com/caiogondim/logdown.js/blob/b296c732a11a91d14df0a8a9c379a5eb668b5601/package.json#L49). Furthermore, the package.json file is just a JSON file, which has __required__ values, not __allowed__ values. So it is expendable with custom values.

> Can you give some input why it is advisable? 

The [official TypeScript docs](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) specify that it would be "advisable" to add the `types` field, although there are no reasons given why.

I see your point following *Convention over Configuration* and if it's really important for you, I will remove the `types` property.

Reply:
Great. Thank you for sharing your point of view. I agree on the fact that the `jest` property should become a `.jestrc` file but that's up to another PR. Regarding the "types" property: In the past Microsoft named it "typings", so that shows that those properties need to be maintained if they are not resolved automatically which is why I prefer *Convention over Configuration* in this case.